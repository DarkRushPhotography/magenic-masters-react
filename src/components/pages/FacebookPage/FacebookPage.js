import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

import Fab from "@material-ui/core/Fab";

import PageHeader from "../../@shared/PageHeader/PageHeader";
import Page from "../../@shared/Page/Page";
import { actionCreators as tinyPngActionCreators } from "../../../store/TinyPng";

class FacebookPage extends Component {
  render() {
    return (
      <Page>
        <PageHeader />
        <h1>Facebook</h1>
        <Fab color="primary" onClick={() => this.props.actions.resizeImage()}>
          RESIZE IMAGE
        </Fab>
        <p>Is Error: {this.props.isError?"Yes":"No"}</p>
        <p>Error Message: {this.props.errorMessage}</p>
      </Page>
    );
  }
}

FacebookPage.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  resizedImage: PropTypes.object,
  isError: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string,
  actions: PropTypes.shape({
    resizeImage: PropTypes.func.isRequired
  })
};

const mapStateToProps = state => ({
  isLoading: state.tinyPng.isLoading,
  resizedImage: state.tinyPng.resizedImage,
  isError: state.tinyPng.isError,
  errorMessage: state.tinyPng.errorMessage
});

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators(
      {
        resizeImage: tinyPngActionCreators.resizeImage
      },
      dispatch
    )
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FacebookPage);
