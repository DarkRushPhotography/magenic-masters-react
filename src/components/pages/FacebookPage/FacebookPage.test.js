import React from "react";
import { shallow } from "enzyme";

import { FacebookPage } from "./FacebookPage";

describe("FacebookPage", () => {
  it("should render correctly", () => {
    const component = shallow(<FacebookPage />);

    expect(component).toMatchSnapshot();
  });
});
