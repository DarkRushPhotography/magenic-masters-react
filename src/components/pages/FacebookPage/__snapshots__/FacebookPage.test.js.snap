// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookPage should render correctly 1`] = `
<Page>
  <withRouter(WithStyles(PageHeader)) />
  <h1>
    Facebook
  </h1>
  <WithStyles(Fab)
    color="primary"
    onClick={[Function]}
  >
    RESIZE IMAGE
  </WithStyles(Fab)>
  <p>
    Is Error: 
    No
  </p>
  <p>
    Error Message: 
  </p>
</Page>
`;
